src/accel/detail/HitProcessor.cc

```diff
@@ -230,26 +273,21 @@ void HitProcessor::operator()(DetectorStepOutput const& out) const
 
         if (navi_)
         {
-            CELER_ASSERT(out.detector[i] < detector_volumes_.size());
+            CELER_ASSERT(out.detector[i] < detectors_.size());
             bool success = this->update_touchable(
```

Comment:
This seems very expensive. Do we really need this check?

Reply:
The touchable update is necessary for detectors like tilecal and hgcal that use the detailed navigation state to determine a subdetector identifier. Since their sensitive detectors require the navigator to be initialized and correct, we don't really have a choice if we're calling back to those routines. My hope is that for our initial implementation this won't completely kill the performance, and doing it this way will give us a "baseline" performance number to justify the effort of implementing the actual detector logic on GPU.

Reply:
I still do not understand why HitProcess is responsible for updating the navigation state (touchable) as a hit is processed at the end of stepping (only once per each step) - isn't the subdetector a physical/logical volume (so can be a tracking volume  with a boundary)?  Isn't it an independent readout channel?  I guess that there may be a confusion between what simulation information needs to be known (for MC "study") and what need to be recorded as hit (to be used for digitization).

Reply:
I might be misunderstanding what the confusion is too, so let me try to restate from a high level what's going on.
- At the end of an event on a local thread, the `LocalTransporter` steps through all the offloaded EM tracks until they die.
- Each step calls the (shared) explicit actions using the shared "params" state and the (thread-local) track states.
- The `celeritas::StepCollector` gathers the position and volume ID at the pre-step and the energy deposition at post-step (though the exact selection of outputs can be changed by the user).
- At post-step, the (shared) HitManager takes the thread-local gathered step data, and calls the thread-local HitProcessor.
- The HitProcessor loops through all the hits, translates them into a thread-local `G4Step` objects, uses the VecGeom logical volume ID to look up the `G4VSensitiveDetector`, and calls `sd->Hit(step)`. If the SD needs to query `hit->GetPreStepPoint()->GetTouchableHandle()` then before calling `Hit` we use a thread-local (i.e., owned by HitProcessor) navigator to update the thread-local touchable for the thread-local step.

When I said "subdetector" I meant "readout channel": I guess HGCal is the subdetector, and the codeneeds the navigation state and volume information to figure out what channel it's in.